app/src/main/java/org/mozilla/focus/utils/DownloadReceiver.java

```diff
@@ -9,33 +9,48 @@
 import org.greenrobot.greendao.Property;
 import org.greenrobot.greendao.query.QueryBuilder;
 import org.mozilla.focus.greenDAO.DBUtils;
+import org.mozilla.focus.greenDAO.DownloadInfo;
 import org.mozilla.focus.greenDAO.DownloadInfoEntity;
 import org.mozilla.focus.greenDAO.DownloadInfoEntityDao;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Created by anlin on 31/07/2017.
  */
 
 public class DownloadReceiver extends BroadcastReceiver {
 
+    private static DownloadReceiver mDownloadReceiver;
+    private List<OnCompleteListener> listeners = new ArrayList<>();
+
+    public interface OnCompleteListener{
+        void onCompleted(DownloadInfo downloadInfo);
+    }
+
+    public static DownloadReceiver getDownloadReceiver(){
```

Comment:
This will be broken when there are multiple BrowserFragment instances. Its life cycle should be handled within BrowserFragment only.